Sorting/QuickSort.cpp

```diff
@@ -1,11 +1,16 @@
 /*
     Quick sort
     ----------
-    An efficient, comparison-based, in-place, divide and conquer sorting algorithm.
+    An efficient, comparison-based, in-place, divide and conquer sorting algorithm. It picks
+    an element as pivot, partitions the given array around the picked pivot and then recursively
+    calls itself on those partitions.
```

Comment:
It randomly picks an element as pivot, partitions the given array around the selected pivot while placing it in its proper position, and recursively calls itself on those partitions.